modules/nf-core/concoct/cutupfasta/main.nf

```diff
@@ -0,0 +1,40 @@
+
+process CONCOCT_CUTUPFASTA {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::concoct=1.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/concoct:1.1.0--py38h7be5676_2':
+        'quay.io/biocontainers/concoct:1.1.0--py38h7be5676_2' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    val(bed)
+
+    output:
+    tuple val(meta), path("*.fasta")                , emit: fasta
+    tuple val(meta), path("*.bed")  , optional: true, emit: bed
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args       = task.ext.args ?: ''
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def bedfile = bed ? "-b ${prefix}.bed" : ""
+    if ("$fasta" == "${prefix}.fasta") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
```

Comment:
You could also catch this by using `stageAs` to stage the input in a subdir

Reply:
Ooh clever... I'll stick to this now as per guidelines but I might use that trick in the future